src/main/kotlin/org/jetbrains/tinygoplugin/ui/TinyGoSettingsUI.kt

```diff
@@ -0,0 +1,80 @@
+package org.jetbrains.tinygoplugin.ui
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory
+import com.intellij.openapi.ui.ComboBox
+import com.intellij.openapi.ui.TextFieldWithBrowseButton
+import com.intellij.ui.EnumComboBoxModel
+import com.intellij.ui.components.JBLabel
+import com.intellij.ui.components.JBTextField
+import com.intellij.util.ui.FormBuilder
+import org.jetbrains.tinygoplugin.GarbageCollector
+import org.jetbrains.tinygoplugin.Scheduler
+import java.awt.event.ActionListener
+import javax.swing.JButton
+import javax.swing.JPanel
+
+class TinyGoSettingsUI(buttonAction: ActionListener) {
```

Comment:
You might have a look at Kotlin UI DSL https://plugins.jetbrains.com/docs/intellij/kotlin-ui-dsl.html